uint8 id
bool permitFiring